internal/storage/detector.go

```
@@ -0,0 +1,83 @@
+
+func (d *CloudDetector) probeGCP(ctx context.Context) bool {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, d.gcpMetadataURL, nil)
+	if err != nil {
+		return false
+	}
+	req.Header.Set("Metadata-Flavor", "Google")
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)